From 9ff8fc326b3c301410fccb7de6465733bdc6d2ed Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Mon, 18 Jun 2018 16:48:19 +0300
Subject: [PATCH 0662/1051] arm64: dts: marvell: add AP806 & CP110 AXI monitors
 description

AP806 has 4 AXI monitors (AXIM v4.0 2.0), with profiling support.
added dts description, with profiling support (required hclk):
- rw/wr monitors on DDR
- rw/wr monitors on South-Bridge interconnect

CP110 has 6 AXI monitors (AXIM v4.0 1.0).
- rw/wr monitors on CP
- rw/wr monitors on PPv2 unit
- rw/wr monitors on IHB interconnet

Change-Id: I401e4f0bf96c22e699219d471ca1917a418ca963
Signed-off-by: Omri Itach <omrii@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi | 37 +++++++++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi | 54 +++++++++++++++++++
 2 files changed, 91 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 1d4bb514e229..a7001df88222 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -172,6 +172,43 @@
 					   <GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>;
 			};
 
+			axim-ddr-rd@840000 {
+				compatible = "marvell,coresight-axim", "arm,primecell";
+				reg = <0x840000 0x1000>;
+				clocks = <&ap_clk 3>, <&ap_clk 5>;
+				clock-names = "apb_pclk", "hclk";
+				bus-width = <40>;
+				port@0 { };
+			};
+
+			axim-ddr-wr@841000 {
+				compatible = "marvell,coresight-axim", "arm,primecell";
+				reg = <0x841000 0x1000>;
+				clocks = <&ap_clk 3>, <&ap_clk 5>;
+				clock-names = "apb_pclk", "hclk";
+				bus-width = <40>;
+				port@0 { };
+			};
+
+			axim-sb-rd@848000 {
+				compatible = "marvell,coresight-axim", "arm,primecell";
+				reg = <0x848000 0x1000>;
+				clocks = <&ap_clk 3>, <&ap_clk 5>;
+				clock-names = "apb_pclk", "hclk";
+				bus-width = <40>;
+				port@0 { };
+			};
+
+			axim-sb-wr@849000 {
+				compatible = "marvell,coresight-axim", "arm,primecell";
+				reg = <0x849000 0x1000>;
+				clocks = <&ap_clk 3>, <&ap_clk 5>;
+				clock-names = "apb_pclk", "hclk";
+				bus-width = <40>;
+				port@0 { };
+			};
+
+
 			odmi: odmi@300000 {
 				compatible = "marvell,odmi-controller";
 				interrupt-controller;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 9f80b7ad8c1d..4678caefc5ee 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -182,6 +182,60 @@
 			interrupts = <ICU_GRP_NSR 77 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
+		CP110_LABEL(axim_cp_rd): axim-cp-rd@3c5000 {
+			compatible = "marvell,coresight-axim", "arm,primecell";
+			reg = <0x3c5000 0x1000>;
+			clocks = <&CP110_LABEL(clk) 1 3>;
+			clock-names = "apb_pclk";
+			bus-width = <40>;
+			port@0 { };
+		};
+
+		CP110_LABEL(axim_cp_wr): axim-cp-wr@3c6000 {
+			compatible = "marvell,coresight-axim", "arm,primecell";
+			reg = <0x3c6000 0x1000>;
+			clocks = <&CP110_LABEL(clk) 1 3>;
+			clock-names = "apb_pclk";
+			bus-width = <40>;
+			port@0 { };
+		};
+
+		CP110_LABEL(axim_ppv2_rd): axim-ppv2-rd@3c0000 {
+			compatible = "marvell,coresight-axim", "arm,primecell";
+			reg = <0x3c0000 0x1000>;
+			clocks = <&CP110_LABEL(clk) 1 3>;
+			clock-names = "apb_pclk";
+			bus-width = <40>;
+			port@0 { };
+		};
+
+		CP110_LABEL(axim_ppv2_wr): axim-ppv2-wr@3c1000 {
+			compatible = "marvell,coresight-axim", "arm,primecell";
+			reg = <0x3c1000 0x1000>;
+			clocks = <&CP110_LABEL(clk) 1 3>;
+			clock-names = "apb_pclk";
+			bus-width = <40>;
+			port@0 { };
+		};
+
+		CP110_LABEL(axim_hb1_rd): axim-hb1-rd@3c8000 {
+			compatible = "marvell,coresight-axim", "arm,primecell";
+			reg = <0x3c8000 0x1000>;
+			clocks = <&CP110_LABEL(clk) 1 3>;
+			clock-names = "apb_pclk";
+			bus-width = <40>;
+			port@0 { };
+		};
+
+		CP110_LABEL(axim_hb1_wr): axim-hb1-wr@3c9000 {
+			compatible = "marvell,coresight-axim", "arm,primecell";
+			reg = <0x3c9000 0x1000>;
+			clocks = <&CP110_LABEL(clk) 1 3>;
+			clock-names = "apb_pclk";
+			bus-width = <40>;
+			port@0 { };
+		};
+
 		CP110_LABEL(thermal): thermal@400078 {
 			compatible = "marvell,armada-cp110-thermal";
 			reg = <0x400078 0x4>,
-- 
2.17.1

